src/test/kotlin/org/neo4j/graphql/utils/CypherTestSuite.kt

```diff
@@ -2,107 +2,108 @@ package org.neo4j.graphql.utils
+
+            val ctx = config ?: QueryContext()
+            Translator(SchemaBuilder.buildSchema(schema))
+                .translate(request, requestParams, ctx)
+                .first()
+        }
+        return {
+            transformationTask.run()
+            transformationTask.get()
         }
     }
 
-    fun translate(query: String, requestParams: Map<String, Any?> = emptyMap(), ctx: QueryContext = QueryContext()): Cypher {
-        return Translator(SchemaBuilder.buildSchema(schema))
-            .translate(query, requestParams, ctx)
-            .first()
+    private fun printGeneratedQuery(result: () -> Cypher): DynamicTest = DynamicTest.dynamicTest("Generated query") {
+        println(result().query)
     }
 
-    fun run(contextProvider: ((requestParams: Map<String, Any?>) -> QueryContext)? = null): List<DynamicTest> {
-        return tests.map {
-            DynamicTest.dynamicTest(it.title,
-                    UriBuilder.fromUri(it.file.toURI()).queryParam("line", it.line).build()) { it.run(contextProvider) }
+    private fun printReplacedParameter(result: () -> Cypher): DynamicTest = DynamicTest.dynamicTest("Generated query with params replaced") {
+        var queryWithReplacedParams = result().query
+        result().params.forEach { (key, value) ->
+            queryWithReplacedParams = queryWithReplacedParams.replace("$$key", if (value is String) "'$value'" else value.toString())
+        }
+        println()
+        println("Generated query with params replaced")
+        println("------------------------------------")
+        println(queryWithReplacedParams)
+    }
+
+    private fun testCypher(title: String, cypherBlock: ParsedBlock?, result: () -> Cypher): DynamicTest = DynamicTest.dynamicTest("Test Cypher", cypherBlock?.uri) {
+        val cypher = cypherBlock?.code?.trim()?.toString()
+                ?: throw IllegalStateException("missing cypher query for $title")
+        Assertions.assertEquals(cypher.normalize(), result().query.normalize())
+    }
+
+    private fun testCypherParams(codeBlocks: Map<String, ParsedBlock>, result: () -> Cypher): DynamicTest {
+        val cypherParamsBlock = codeBlocks["[source,json]"]
+        return DynamicTest.dynamicTest("Test Cypher Params", cypherParamsBlock?.uri) {
+            val resultParams = result().params
+            val cypherParams = cypherParamsBlock?.code?.toString()?.parseJsonMap() ?: emptyMap()
+            Assertions.assertEquals(fixNumbers(cypherParams), fixNumbers(resultParams)) {
+                "\nExpected : ${MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(cypherParams)}\n" +
+                        "Actual   :${MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(resultParams)}"
+            }
         }
     }
 
     companion object {
-        val DEBUG = false
```

Comment:
should come from an external config or system property so it can be changed externally for test runs